BudgetBuddy/src/main/java/com/budgetbuddy/dto/Bill.java

```diff
@@ -6,12 +6,14 @@
 import jakarta.persistence.Id;
 import lombok.Data;
 
+import java.time.LocalDate;
+
 @Entity
 public @Data class Bill {
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private int billID;
     private double billAmount;
-    private String billDueDate;
+    private LocalDate billDueDate;

```

Comment:
It is better to use a date datatype for dates so you dont have to convert them.
https://www.baeldung.com/java-clean-code